modules/transformation.py

```diff
@@ -0,0 +1,500 @@
+                page['_document_height']) > 0 else 0,
+            'document_width': page['_document_width'] if page.get('_document_width') and int(
+                page['_document_width']) > 0 else 0,
+            'localstorage_size': page['_localstorage_size'] if page.get('_localstorage_size') and int(
+                page['_localstorage_size']) > 0 else 0,
+            'sessionstorage_size': page['_sessionstorage_size'] if page.get('_sessionstorage_size') and int(
+                page['_sessionstorage_size']) > 0 else 0,
+            'meta_viewport': page.get('_meta_viewport'),
+            'num_iframes': page.get('_num_iframes'),
+            'num_scripts': page.get('_num_scripts'),
+            'num_scripts_sync': page.get('_num_scripts_sync'),
+            'num_scripts_async': page.get('_num_scripts_async'),
+            'usertiming': page.get('_usertiming')
+        }
+
+    @staticmethod
+    def aggregate_stats(entries, page_id, first_url, first_html_url, status_info):
+        # CVSNO - move this error checking to the point before this function is called
+        if not first_url:
+            logging.error("ERROR($gPagesTable pageid: {}}): no first URL found.", page_id)
+            return
+        if not first_html_url:
+            logging.error("ERROR($gPagesTable pageid: {}): no first HTML URL found.", page_id)
+            return
+
+        # initialize variables for counting the page's stats
+        bytes_total = 0
+        req_total = 0
+        size = {}
+        count = {}
+
+        # This is a list of all mime types AND file formats that we care about.
+        typs = ["css", "image", "script", "html", "font", "other", "audio", "video", "text", "xml", "gif", "jpg", "png",
+                "webp", "svg", "ico", "flash", "swf", "mp4", "flv", "f4v"]
+        # initialize the hashes
+        for typ in typs:
+            size[typ] = 0
+            count[typ] = 0
+        domains = {}
+        maxage_null = max_age_0 = max_age_1 = max_age_30 = max_age_365 = max_age_more = 0
+        bytes_html_doc = num_redirects = num_errors = num_glibs = num_https = num_compressed = max_domain_reqs = 0
+
+        for entry in entries:
+            url = entry['urlShort']
+            pretty_type = entry['type']
+            resp_size = int(entry['respSize'])
+            req_total += 1
+            bytes_total += resp_size
+            count[pretty_type] += 1
+            size[pretty_type] += resp_size
+
+            frmt = entry.get('format')
+            if frmt and pretty_type in ['image', 'video']:
+                if frmt not in typs:
+                    logging.warning(f"Unexpected type, found format:{frmt}, status_info:{status_info}")
+                else:
+                    count[frmt] += 1
+                    size[frmt] += resp_size
+
+            # count unique domains (really hostnames)
+            matches = re.findall(r'http[s]*://([^/]*)', url)
```

Comment:
Do we use any other schemes? E.g. `ws://` and `wss://` for web sockets?

Reply:
I think it's ok to leave this as-is for migration purposes, and consider leaving a TODO or open an issue to improve our classification.